sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Utils.java

```diff
@@ -646,4 +648,23 @@ public static String toJson(ObjectMapper mapper, ObjectNode object) {
             throw new IllegalStateException("Unable to convert JSON to STRING", e);
         }
     }
+
+    public static String limitedStackTrace(Exception e, int depth) {
+        StackTraceElement[] stacktrace = e.getStackTrace();
+        if (stacktrace == null) {
+            return "null";
+        } else {
+            StringWriter strWriter = new StringWriter();
+            PrintWriter writer = new PrintWriter(strWriter);
```

Comment:
We should put these in a try with finally block to close both StringWriter and PrintWriter in case there is any issue serializing the exception. 
Just as a safety check so that we don't have memory leaks.

Reply:
This is not a resource. Using of StringBuilder is the same of using byte[] or StringBuilder. same as byte[] instantiation that is not need to be explicilty closed, doesn't cause memory leak.